modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -1222,6 +1222,34 @@ CV__DNN_INLINE_NS_BEGIN
         CV_PROP_RW DataLayout datalayout; //!< Order of output dimensions. Choose DNN_LAYOUT_NCHW or DNN_LAYOUT_NHWC.
         CV_PROP_RW ImagePaddingMode paddingmode;   //!< Image padding mode. @see ImagePaddingMode.
         CV_PROP_RW Scalar borderValue;   //!< Value used in padding mode for padding.
+
+        /** @brief Get rectangle coordinates in original image system from rectangle in blob coordinates.
+         *  @param rBlob rect in blob coordinates.
+         *  @param sizeOri original input image size inserted in blob.
+         *  @returns rectangle in original image coordinates.
+         */
+        CV_WRAP Rect blobRectToImageRect(const Rect &rBlob, const Size &sizeOri);
+
+        /** @brief Get rectangle coordinates in original image system from rectangle in blob coordinates.
+         *  @param rBlob rect in blob coordinates.
+         *  @param rImg result rect in image coordinates.
+         *  @param size spatial size for output image
+         */
+        CV_WRAP void blobRectsToImageRects(const std::vector<Rect> &rBlob, CV_OUT std::vector<Rect>& rImg, const Size& size);
+
+        /** @brief Get rectangle coordinates in original image system from rectangle in blob coordinates.
+         *  @param rBlob rect in blob coordinates.
+         *  @param rImg result rect in image coordinates.
+         *  @param size spatial size for output image
+         */
+        CV_WRAP_AS(blobRectsToImageRects2d) void blobRectsToImageRects (const std::vector<Rect2d> &rBlob, CV_OUT std::vector<Rect2d>& rImg, const Size& size);
```

Comment:
@LaurentBerger , may we ask to remove Rect2d from this PR and if it's a critical feature, discuss separately? In my point of view, users mostly work on Rect (as a ROI or drawable bounding box).

Reply:
In NMSBoxes there is three type Rect, Rect2d  and RotatedRect. Now in blobRect(s)ToImageRect(s) there is a scale factor error can be 0 to 2*scaleFactor pixels (2 because there is origin point and dimension)

Reply:
Because NMSBoxes might be applied as for relative Rect2d (values from 0 to 1) as for absolute pixel coordinates. On blobToImageRect we always return pixel coordinates and subpixel precision is redundant.

Reply:
After this [one](https://github.com/opencv/opencv/pull/24173) and [this ](https://github.com/opencv/opencv/pull/24267) there is now this

Reply:
Let me add a commit to demonstrate my vision of the version with Rect2d.